mesonbuild/compilers/vala.py

```diff
@@ -119,3 +140,24 @@ def thread_flags(self, env: 'Environment') -> T.List[str]:
 
     def thread_link_flags(self, env: 'Environment') -> T.List[str]:
         return []
+
+    def get_option_link_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
+        return []
+
+    def links(self, code: 'mesonlib.FileOrString', env: 'Environment', *,
+              compiler: T.Optional['Compiler'] = None,
+              extra_args: T.Union[None, T.List[str], CompilerArgs, T.Callable[[CompileCheckMode], T.List[str]]] = None,
+              dependencies: T.Optional[T.List['Dependency']] = None,
+              disable_cache: bool = False) -> T.Tuple[bool, bool]:
+        self.force_link = True
```

Comment:
I don't love this, but I don't see a better way to handle this problem :/

Reply:
Yep, me too.